OpenRA.Mods.Common/Activities/PickupUnit.cs

```diff
@@ -88,7 +88,8 @@ public override bool Tick(Actor self)
 			if (cargo.IsDead || carryable.IsTraitDisabled || !cargo.AppearsFriendlyTo(self))
 			{
 				carryall.UnreserveCarryable(self);
-				return true;
+				Cancel(self, true);
+				return false;
```

Comment:
This does nothing. You need to set ChildHasPriority = true; before returning or the Tick won't be run again

Reply:
Can you explain to me why? Returning false means that the activity didn't complete, so we will run into the `IsCanceling` branch.

Reply:
Cancel, cancels the child activity, and when child activity is null and base activity has ChildHasPriority set to false, the activity won't receive a tick

Reply:
If the child doesn't have priority we run into the else branch and tick:
https://github.com/OpenRA/OpenRA/blob/06edc3dff11b0fba891f694b1452dc3d08bd99c7/OpenRA.Game/Activities/Activity.cs#L112-L120

Reply:
The tick doesn't run. You can debug more it if you want, I have done that plenty.

For the best tescase use the ts caryall, when caryall is decending to pickup, kill the unit its trying to pickup